tests/acceptance/steps/user-identity-credentials.steps.ts

```diff
@@ -0,0 +1,41 @@
+import { Given } from "@cucumber/cucumber";
+import { WorldDefinition } from "./base-verbs.step";
+import { evcsPatchCredentials } from "./utils/evcs-api";
+import { createAndPostCredentials, createAndPostFraudCheckCredential } from "./helpers/credential-helpers";
+
+Given<WorldDefinition>("a user has {int} CURRENT credentials stored", async function (credentials: number) {
+  this.credentialJwts = await createAndPostCredentials(credentials, this.userId);
+});
+
+Given<WorldDefinition>("an extra CURRENT credential is stored for the user", async function () {
+  await createAndPostCredentials(1, this.userId);
+});
+
+Given<WorldDefinition>("an existing CURRENT credential is marked as HISTORIC for the user", async function () {
+  if (!this.credentialJwts?.length) {
+    throw new Error("The step expects to be able to access credentials from the WorldDefinition");
+  }
+
+  const signatureOfCredentialToUpdate = this.credentialJwts!.at(0)?.split(".").at(-1) as string;
+  await evcsPatchCredentials(this.userId, [{ signature: signatureOfCredentialToUpdate, state: "HISTORIC" }]);
+});
+
+Given<WorldDefinition>(
+  "the user has a fraud check credential with nbf {int} months ago",
+  async function (months: number) {
+    const nbfDate = new Date();
+    nbfDate.setDate(nbfDate.getDate() - months * 30);
+    const fraudCheckJwt = await createAndPostFraudCheckCredential(this.userId, nbfDate);
+    this.credentialJwts.push(fraudCheckJwt);
+  }
+);
+
+Given<WorldDefinition>(
+  "the user has a fraud check credential with nbf {int} months ago and fraudCheck {string}",
```

Comment:
I'm wondering if we could make it more obvious that the "fraudCheck" mentioned in the step definition is in the `failedCheckDetails` (unsuccessful check). Because in theory it's also possible for "fraudCheck" to exist on `checkDetails` (successful check).

Maybe we reword slightly to something like:

"the user has a fraud check credential with nbf {int} months ago and failed fraudCheck {string}"

or

"the user has an unsuccessful fraud check credential with nbf {int} months ago and fraudCheck {string}"

Reply:
Good suggestion - agreed! Have gone with your first option, thanks for explaining